### ISSUE TITLE
Light-dark-theme-switch icon looks not aligned vertically.

### ISSUE THREAD

**Issue by Snoopy1866** [executablebooks/MyST-Parser#995]
### Describe the gap in the current documentation

- `MyST`'s icon: 
![image](https://github.com/user-attachments/assets/2f458cc2-9f75-48fd-a960-87b75adf27db)

- `sphinx_book_theme`'s icon (It's better!):
![image](https://github.com/user-attachments/assets/486a1027-7ad5-4d8d-9cd6-849fa4e46462)

### Describe the solution you'd like

I don't know why this would happed, since the html theme of MyST's documentation is exactly `sphinx_book_theme`

**Comment by Snoopy1866** [comment]
The icon also looks not aligned vertically in the **latest** version of `sphinx_book_theme`'s documentation, **stable** version is OK.

**Comment by agoose77** [comment]
Good catch @Snoopy1866! The cause for this is an upstream styling change in pydata-sphinx-theme which needs to be corrected in sphinx-book-theme.
